test/common/index.js

```diff
@@ -723,6 +683,27 @@ function runWithInvalidFD(func) {
   printSkipMessage('Could not generate an invalid fd');
 }
 
+function requireFlags(flags) {
+  if (!(flags instanceof Array)) {
+    flags = [flags];
+  }
+  let missing = flags.filter((flag) => !process.execArgv.includes(flag));
+
+  // Special handling for worker_threads. Can be removed once worker_threads
+  // are no longer behind a flag.
+  if (missing.length === 1 &&
```

Comment:
Without the check, some tests fail with `tools/test.py --worker`. However, I now realize I should be using `isMainThread` rather than a length check. It is probably more robust and regardless it makes it more obvious what is going on. So I'll make that change.